widgets/InputField/InputField.bbj

```diff
@@ -1,282 +1,149 @@
-        #passwordField!.addStyle("inputFieldStyleInput")
-        #passwordField!.setLeftMargin(0)
-        #passwordField!.setPasswordVisible(0)
-        #passwordField!.setCallback(BBjAPI.ON_GAINED_FOCUS,#this!,"onFocusGained")
-        #passwordField!.setCallback(BBjAPI.ON_LOST_FOCUS,#this!,"onFocusLost")
-        #passwordField!.setCallback(BBjAPI.ON_EDIT_MODIFY,#this!,"onChange")
-    methodend
-    
+    rem ||||||||||||||||||||||||||||||||||||||||||||
+    rem |||||||||||||||| CALLBACKS |||||||||||||||||
+    rem ||||||||||||||||||||||||||||||||||||||||||||
+ 
     method private void createIcon()
         #fieldIcon! =  #setIcon(#window!, #suffixIconPath!)
         #fieldIcon!.addStyle("inputFieldIcon")
     methodend
 
-    rem ||||||| CALLBACKS ||||||||
     method public void onHoverStart(BBjMouseEnterEvent event!)
         #label!.addStyle("hover")
     methodend
     
     method public void onHoverExit(BBjMouseExitEvent event!)
         #label!.removeStyle("hover")
     methodend
-    
+
     method public void onFocusGained(BBjGainedFocusEvent event!)
         #label!.addStyle("focus")
-        
-        if #isPasswordField! then
-            #passwordField!.addStyle("focus")
-        else
-            #inputField!.addStyle("focus")   
-        endif
+        #inputField!.addStyle("focus")   
     methodend
     
     method public void onFocusLost(BBjLostFocusEvent event!)
         #label!.removeStyle("focus")
-        if #isPasswordField! then
-            #passwordField!.removeStyle("focus")
-        else
-            #inputField!.removeStyle("focus")
-        endif
-    methodend
-
-    method public void onError()
-        #label!.setStyle("color","var(--webkit-primary)")
-        if #isPasswordField! then
-            #passwordField!.addStyle("error")
-        else
-            #inputField!.addStyle("error")
-        endif
+        #inputField!.removeStyle("focus")
     methodend
 
-    method public void onChange(BBjEditModifyEvent event!)
-        inputText! = event!.getText()
-
-        if inputText!.length() = 1 then 
-            #dialog!.clearStyles()
-            #label!.addStyle("focus")
-            if #isPasswordField! then
-                #passwordField!.addStyle("focus")
-            else
-                #inputField!.addStyle("focus")    
+    method public void onInputChange(BBjEditModifyEvent event!)  
+        if event!.getText() = "" then 
+            #errorMessage!.setText("") 
+        else 
+            if #validator! <> null() then
+                #errorMessage!.setText(#validator!.validate(#inputField!.getText()))
+                if #errorMessage!.getText() <> "" then 
+                    #inputField!.addStyle("errorMessage")
+                    #valid! = Boolean.FALSE 
+                else 
+                    #inputField!.removeStyle("errorMessage")
+                    #valid! = Boolean.TRUE 
+                endif
             endif
```

Comment:
`onInputChange` shoule fire an custom event, so that top level component can implement custom logic for onChange.
Use case - 
In form control, we need to bind input fields to data row's fields. But in current implementation no event is fired from `InputField` on value change. Therefore there's no way to handle change in form control and update data row's field value.

Reply:
Looks good to me for the most part, all functionality seems to work. However it loads quite differently in the Prodin compared to other branches
![logindialog](https://user-images.githubusercontent.com/90770238/140027096-256a413c-fb48-4bea-8e51-157a8401cc54.png)
: